### PR TITLE
[NO-JIRA] Fix debug view hierarchy

### PR DESCRIPTION
**Pull request by georgegillams**
Instead of hackily replacing the window to enable shake gesture, I've subclassed UIApplication and added the logic there.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Example/Backpack/main.m [hunk 1]
```diff
@@ -22,6 +22,6 @@
 int main(int argc, char * argv[])
 {
     @autoreleasepool {
-        return UIApplicationMain(argc, argv, nil, NSStringFromClass([BPKAppDelegate class]));
+        return UIApplicationMain(argc, argv, @"BPKExampleApp", NSStringFromClass([BPKAppDelegate class]));
```
Comment (k0nserv) [inline Example/Backpack/main.m]: Can we use `NSStringFromClass([BPKExampleApp class])` please?

File: Example/Backpack/BPKExampleApp.m [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ */
+
+#import "BPKExampleApp.h"
+#import <Backpack/Color.h>
+
+@implementation BPKExampleApp
+
+- (void)sendEvent:(UIEvent *)event
+{
```
Comment (k0nserv) [inline Example/Backpack/BPKExampleApp.m]: Can we fix up the code style in this method?

File: Example/Backpack/BPKExampleApp.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+@import UIKit;
```
Comment (k0nserv) [inline Example/Backpack/BPKExampleApp.h]: Can we stick to `#import <UIKit/UIKit.h>` for the time being? We might conclude that we should use the `@import` syntax more, but let's make that an explicit decision if so

File: Example/Backpack/BPKExampleApp.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+@interface BPKExampleApp : UIApplication
```
Comment (k0nserv) [inline Example/Backpack/BPKExampleApp.h]: This is missing `NS_ASUSME_NONNULL_BEGIN`/`NS_ASSUME_NONNULL_END` markers

File: Example/Backpack/BPKExampleApp.h (hunk 3 above)
Comment (shaundon) [reply Example/Backpack/BPKExampleApp.h]: I didn't even know about this syntax!

Comment (k0nserv) [reply Example/Backpack/BPKExampleApp.h]: It was introduce a while back and is specific to frameworks(I think?). See https://stackoverflow.com/questions/18947516/import-vs-import-ios-7

**Pull request merged**